### ISSUE TITLE
Getting distributed training and mixed precision training to work

### ISSUE THREAD

**Issue by dchang56** [allenai/allennlp#4000]
I'm trying to use both distributed training and mixed precision training to speed up a simple bert experiment (much like snli_roberta.json). However, adding them to the config file (individually and at the same time) actually slowed down the training (4-5min per epoch instead of 2-3min). Below is the relevant snippet of the config file.

I'd appreciate any thoughts!

    
    "trainer": {
        "num_epochs": 5,
        "validation_metric": "+accuracy",
        "cuda_device": 0,
        "optimizer": {
            "type": "huggingface_adamw",
            "lr": 3e-5,
            "weight_decay": 0.1
        },
        "learning_rate_scheduler": {
            "type": "slanted_triangular",
            "cut_frac": 0.06
        },
        "opt_level": "O1"
    },
    "distributed": {
        "cuda_devices": [0, 1, 2]
    }

**Comment by dirkgr** [comment]
We have some performance work planned already, but it will likely happen after the 1.0 release.

I would guess that at 2 min per epoch, you are unlikely to gain time by using multiple GPUs. It will spend all its time reading data and copying it to and from the GPU already. Do you even max out one GPU in this configuration?
